src/OSPSuite.Core/Domain/UnitSystem/Dimension.cs

```diff
@@ -166,14 +165,15 @@ public Unit DefaultUnit
 
       public IEnumerable<Unit> Units => _units;
 
-      public IEnumerable<string> GetUnitNames()
-      {
-         return _units.Keys;
-      }
+      public IEnumerable<string> GetUnitNames() => _units.Keys;
 
       public Unit Unit(string name)
       {
-         return _units[name];
+         if (_units.Contains(name))
```

Comment:
@Yuri05 Unit by name. Check first if there is a unit with name (cheap) and return
Otherwise lookup by synonym
I wanted to cache this but decided otherwise because one can add synonym after the fact programatically